app/carbonapi/app.go

```diff
@@ -51,43 +52,62 @@ type App struct {
 	defaultTimeZone *time.Location
 
 	backends []backend.Backend
+
+	prometheusMetrics PrometheusMetrics
 }
 
-var prometheusMetrics = struct {
+// PrometheusMetrics are metrix exported via /metrics endpoint for Prom scraping
+type PrometheusMetrics struct {
 	Requests     prometheus.Counter
 	Responses    *prometheus.CounterVec
 	DurationsExp prometheus.Histogram
 	DurationsLin prometheus.Histogram
-}{
-	Requests: prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Name: "http_requests_total",
-			Help: "Count of HTTP requests",
-		},
-	),
-	Responses: prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "http_responses_total",
-			Help: "Count of HTTP responses, partitioned by return code and handler",
-		},
-		[]string{"code", "handler"},
-	),
-	DurationsExp: prometheus.NewHistogram(
-		prometheus.HistogramOpts{
-			Name:    "http_request_duration_seconds_exp",
-			Help:    "The duration of HTTP requests (exponential)",
-			Buckets: prometheus.ExponentialBuckets((50 * time.Millisecond).Seconds(), 2.0, 20),
-		},
-	),
-	DurationsLin: prometheus.NewHistogram(
-		prometheus.HistogramOpts{
-			Name:    "http_request_duration_seconds_lin",
-			Help:    "The duration of HTTP requests (linear)",
-			Buckets: prometheus.LinearBuckets(0.0, (50 * time.Millisecond).Seconds(), 40), // Up to 2 seconds
-		},
-	),
+	TimeInQueue  prometheus.Histogram
```

Comment:
Why we need Histogram? Why no SummaryVec?

Reply:
Because we want to have more fine-graining and want to use Prometheus primitives.